node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -3747,6 +3939,39 @@ mod tests {
         assert_eq!(dest_db.node_by_key(node_f.public_key()), None);
     }
 
+    fn fix_last_update_nodes(
+        expected_db: &mut NeighborhoodDatabase,
+        dest_db: &NeighborhoodDatabase,
+        root_node: &NodeRecord,
+    ) {
+        expected_db
+            .node_by_key_mut(root_node.public_key())
+            .unwrap()
+            .metadata
+            .last_update = dest_db
+            .node_by_key(root_node.public_key())
+            .unwrap()
+            .metadata
+            .last_update;
```

Comment:
You could write

```Rust
let fix_last_update = |pub_key: &PublicKey|{
    dest_db.node_by_key(pubkey).map(|node_record|
        expected_db
            .node_by_key_mut(pubkey)
            .unwrap()
            .metadata
            .last_update = node_record.metadata.last_update;
        expected_db
            .node_by_key_mut(pubkey)
            .unwrap()
            .resign();
)};

```

Use it like this for the operation in the upper half: 

```Rust

fix_last_update(root_node.pub_key()).expect("did not find root node");

``` 
and like this in the for_each() loop:

```Rust

  keys.into_iter() .for_each(|pubkey| {let _ = fix_last_update(pubkey);});

```


My formatting is awful because I write it without any tools. :-)